tests/kubernetes-e2e.rs

```diff
@@ -0,0 +1,725 @@
+    let _guard = lock();
+    let framework = make_framework();
+
+    let vector = framework.vector("test-vector", VECTOR_CONFIG).await?;
+    framework
+        .wait_for_rollout("test-vector", "daemonset/vector", vec!["--timeout=30s"])
+        .await?;
+
+    let test_namespace = framework.namespace("test-vector-test-pod").await?;
+
+    let test_message = generate_long_string(8, 8 * 1024); // 64 KiB
+    let test_pod = framework
+        .test_pod(test_pod::Config::from_pod(&make_test_pod(
+            "test-vector-test-pod",
+            "test-pod",
+            &format!("echo {}", test_message),
+            vec![],
+        ))?)
+        .await?;
+    framework
+        .wait(
+            "test-vector-test-pod",
+            vec!["pods/test-pod"],
+            WaitFor::Condition("initialized"),
+            vec!["--timeout=30s"],
+        )
+        .await?;
+
+    let mut log_reader = framework.logs("test-vector", "daemonset/vector")?;
+    smoke_check_first_line(&mut log_reader).await;
+
+    // Read the rest of the log lines.
+    let mut got_expected_line = false;
+    look_for_log_line(&mut log_reader, |val| {
+        if val["kubernetes"]["pod_namespace"] != "test-vector-test-pod" {
+            // A log from something other than our test pod, pretend we don't
+            // see it.
+            return FlowControlCommand::GoOn;
+        }
+
+        // Ensure the message we got matches the one we emitted.
+        assert_eq!(val["message"], test_message);
+
+        if got_expected_line {
+            // We've already seen our expected line once! This is not good, we
+            // only emitted one.
+            panic!("test message seen more than once");
+        }
+
+        // If we did, remember it.
+        got_expected_line = true;
+
+        // Request to stop the flow.
+        FlowControlCommand::Terminate
+    })
+    .await?;
+
+    assert!(got_expected_line);
+
+    drop(test_pod);
+    drop(test_namespace);
+    drop(vector);
+    Ok(())
+}
+
+/// This test validates that vector picks up preexisting logs - logs that
+/// existed before vector was deployed.
+#[tokio::test]
+async fn preexisting() -> Result<(), Box<dyn std::error::Error>> {
+    let _guard = lock();
+    let framework = make_framework();
+
+    let test_namespace = framework.namespace("test-vector-test-pod").await?;
+
+    let test_pod = framework
+        .test_pod(test_pod::Config::from_pod(&make_test_pod(
+            "test-vector-test-pod",
+            "test-pod",
+            "echo MARKER",
+            vec![],
+        ))?)
+        .await?;
+    framework
+        .wait(
+            "test-vector-test-pod",
+            vec!["pods/test-pod"],
+            WaitFor::Condition("initialized"),
+            vec!["--timeout=30s"],
+        )
+        .await?;
+
+    // Wait for some extra time to ensure pod completes.
+    tokio::time::delay_for(std::time::Duration::from_secs(10)).await;
```

Comment:
It would be nice to wait for `Succeeded` pod phase instead of a delay, or maybe in conjuction if there is a race with other parts of the system, like filesystem and container runtime.

Reply:
I didn't find a way to wait for `Succeesed` state of a raw `Pod` (the one not in a `Job`). If you can show me `kubectl` command example - that would be great!
For now, tests rely on waiting for `initialized` state, and that test, in particular, assumes the `echo` in a `Pod` completes within 10 seconds. It's a race, and it would definitely be better to have it work based on explicit synchronization. I'll take a look to see if there's any state change we rely on to improve it.

Reply:
True, I was still thinking in the context of `http` api. Shame `kubectl` doesn't have this.

Reply:
I thougt about it. Technically it’s doable, and it’s not that much much work. One provlem is, though, the solution would itself be racy. I’m thinking about calling `kubectl -ogo -f '{{...}}'` in a loop until we see the status we’re looking for. To safeguard from the races, we’d have to restrict the API such that we can only wait for certtain  statuses, instead of the arbitrary ones.

Reply:
I created #3150 so we don't lose track of it.